probatus/feature_elimination/feature_elimination.py

```diff
@@ -695,3 +680,245 @@ def plot(self, show=True, **figure_kwargs):
+        iteration, the model is fitted on the train folds, and applied on the validation fold to estimate
+        SHAP feature importance. The model is trained until the scoring metric, measured on the validation fold,
+        stops improving after a number of early_stopping_rounds.
+    3. Remove `step` lowest SHAP importance features from the dataset.
+
+    At the end of the process, the user can plot the performance of the model for each iteration, and select the
+        optimal number of features and the features set.
+
+    The functionality is
+        similar to [RFECV](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.RFECV.html).
+        The main difference is removing the lowest importance features based on SHAP features importance. It also
+        supports the use of sklearn compatible search CV for hyperparameter optimization e.g.
+        [GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LassoCV.html),
+        [RandomizedSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html?highlight=randomized#sklearn.model_selection.RandomizedSearchCV), or
+        [BayesSearchCV](https://scikit-optimize.github.io/stable/modules/generated/skopt.BayesSearchCV.html), which
+        needs to be passed as the `clf`. Thanks to this you can perform hyperparameter optimization at each step of
+        the feature elimination. Lastly, it supports categorical features (object and category dtype) and missing values
+        in the data, as long as the model supports them.
+
+    We recommend using [LGBMClassifier](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMClassifier.html),
+        because by default it handles missing values and categorical features. In case of other models, make sure to
+        handle these issues for your dataset and consider impact it might have on features importance.
+
+
+    Example:
+    ```python
+    from lightgbm import LGBMClassifier
+    import numpy as np
+    import pandas as pd
+    from probatus.feature_elimination import EarlyStoppingShapRFECV
+    from sklearn.datasets import make_classification
+    from sklearn.model_selection import train_test_split
+    from sklearn.ensemble import RandomForestClassifier
+    from sklearn.model_selection import RandomizedSearchCV
+
+    feature_names = [
+        'f1', 'f2', 'f3', 'f4', 'f5', 'f6', 'f7',
+        'f8', 'f9', 'f10', 'f11', 'f12', 'f13',
+        'f14', 'f15', 'f16', 'f17', 'f18', 'f19', 'f20']
+
+    # Prepare two samples
+    X, y = make_classification(n_samples=200, class_sep=0.05, n_informative=6, n_features=20,
+                               random_state=0, n_redundant=10, n_clusters_per_class=1)
+    X = pd.DataFrame(X, columns=feature_names)
+
+
+    # Prepare model and parameter search space
+    clf = LGBMClassifier(n_estimators=200)
```

Comment:
I think this one is not gonna run correctly on macos github actions. Macos has troubles with running Lightgbm. 

I propose making a separate function for testing docstrings with lightgbm in `tests/docs/test_docstring.py`, with the following fixture:
```
@pytest.mark.skipif(os.environ.get("SKIP_LIGHTGBM") == "true", reason="LightGBM tests disabled")
```
Then the lgbm tests are going to be skipped for mac only.

Possibly the same will be required in `tests/docs/test_notebooks.py`